src/anchored_set_variable.cc

```diff
@@ -109,12 +109,13 @@ void AnchoredSetVariable::resolve(const std::string &key,
 
 std::unique_ptr<std::string> AnchoredSetVariable::resolveFirst(
     const std::string &key) {
-    auto range = equal_range(key);
-    for (auto it = range.first; it != range.second; ++it) {
-        std::unique_ptr<std::string> b(new std::string());
-        b->assign(it->second->getValue());
+
+    if (auto search = this->find(key); search != this->end()) {
+        auto b = std::make_unique<std::string>();
+        b->assign(search->second->getValue());
         return b;
```

Comment:
Not sure, but if you improve this, I think this can be optimized by constructing this inline and avoid the resizing between line 114 and line 115

Probably like (not tested)
```cpp
return std::make_unique<std::string>(search->second->getValue());
```

Reply:
Ah, you are right, this make sense. And yes, looks like it works. Changed in [16d0df0](https://github.com/owasp-modsecurity/ModSecurity/pull/3117/commits/16d0df0ff9598cedec29330944e09151dba8fea2), thanks!